src/modules/teams.js

```diff
@@ -0,0 +1,220 @@
+        throw Error(response.statusText);
+      })
+      .then((response) => {
+        const team = response.data;
+        dispatch({
+          type: GET_TEAM,
+          payload: team
+        });
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        return team;
+      })
+      .catch((error) => {
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        console.warn('error', error)
+      });
+  };
+}
+
+const getBody = (team) => {
+  return JSON.stringify({
+    name: team.name,
+    managers: team.managers.filter(unique),
+    confirmedUsers: team.confirmedUsers.filter(unique),
+    users: team.users.filter(unique),
+    areas: team.areas
+  })
+}
+
+export function createTeam(team) {
+  const url = `${API_BASE_URL}/teams/`;
+  return (dispatch, state) => {
+    return fetch(url, {
+      headers: {
+        Authorization: `Bearer ${state().user.token}`,
+        'Content-Type': 'application/json'
+      },
+      method: 'POST',
+      body: getBody(team)
+    })
+      .then((response) => {
+        if (response.ok) return response.json();
+        throw Error(response.statusText);
+      })
+      .then((response) => {
+        const team = response.data;
+        dispatch({
+          type: SET_EDITING,
+          payload: false
+        });
+        dispatch({
+          type: SET_LOADING,
+          payload: true
+        });
+        dispatch({
+          type: SAVE_TEAM,
+          payload: team
+        });
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        dispatch(getTeam(state().user.data.id));
+        if (state().teams.sendNotifications) {
+          dispatch({
+            type: SEND_NOTIFICATIONS,
+            payload: false
+          });
+          toastr.success('Request sent');
+        }
+        return team;
+      })
+      .catch((error) => {
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        console.warn('error', error)
+      });
+  };
+}
+export function updateTeam(team, id) {
+  const url = `${API_BASE_URL}/teams/${id}`;
+  return (dispatch, state) => {
+    return fetch(url,
+      {
+        headers: {
+          Authorization: `Bearer ${state().user.token}`,
+          'Content-Type': 'application/json'
+        },
+        method: 'PATCH',
+        body: getBody(team)
+      }
+    )
+      .then((response) => {
+        if (response.ok) return response.json();
+        throw Error(response.statusText);
+      })
+      .then((response) => {
+        const team = response.data;
+        dispatch({
+          type: SET_LOADING,
+          payload: true
+        });
+        dispatch({
+          type: SAVE_TEAM,
+          payload: team
+        });
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        dispatch(getTeam(state().user.data.id));
```

Comment:
Is true that the admin should't change right now. Just in case of multiple users using it at the same time